src/App.tsx

```diff
@@ -1,24 +1,57 @@
-import React, { ReactElement } from "react";
+import React, { ReactElement, useState } from "react";
 import Transformation from "./Transformation";
 import { SavedTransformation } from "./transformation-components/types";
 import { initPhone } from "./utils/codapPhone";
+import "./App.css";
 
-// This should be a pure function (ie: only render once)
 export const App = (): ReactElement => {
-  const parsedUrl = new URL(window.location.href);
+  const [loading, setLoading] = useState<boolean>(true);
+  const loadingMsg = <p className="loading">Connecting to CODAP...</p>;
+
+  const [initError, setInitError] = useState<boolean>(false);
+  const initErrorMsg = (
+    <p className="initError">
+      Could not connect to CODAP. Please make sure you are using CODAP Flow
+      within CODAP.
+    </p>
+  );
+
+  /**
+   * Initializes a new data interactive (plugin) with the given name,
+   * and turns off loading once it has succeeded, or indicates an
+   * error should be displayed if it fails.
+   */
+  const initWithPluginName = (name: string): void => {
+    initPhone(name)
+      .then(() => {
+        setLoading(false);
+      })
+      .catch(() => {
+        setInitError(true);
+      });
+  };
 
+  const parsedUrl = new URL(window.location.href);
   const transformation = parsedUrl.searchParams.get("transform");
+  let pluginContent: JSX.Element;
 
   if (transformation === null) {
-    initPhone("CODAP Flow");
-    return <Transformation />;
+    initWithPluginName("CODAP Flow");
```

Comment:
Should we have a file for constants like the plugin name? We can also change this to "Transformers"

Reply:
Probably a good idea--I have made this change to "Transformers" though on the `further-architecture-changes` branch, so maybe we should have that be there.